Tasks/PublishSymbols/Publish-Symbols.ps1

```diff
@@ -0,0 +1,284 @@
+    Write-Host "Running: $exe $displayArgs"
+
+    $processInfo = New-Object System.Diagnostics.ProcessStartInfo
+    $processInfo.FileName = $exe
+    $processInfo.UseShellExecute = $false
+    $processInfo.RedirectStandardOutput = $true
+    $processInfo.RedirectStandardError = $true
+    $processInfo.Arguments = $arguments
+
+    $process = New-Object System.Diagnostics.Process
+    $process.StartInfo = $processInfo
+   
+    $process.Start() | Out-Null
+
+    $stdout = $process.StandardOutput.ReadToEnd()
+    $stderr = $process.StandardError.ReadToEnd()
+
+    $process.WaitForExit()
+
+    $stdout = $stdout | ForEach-Object { $_.Replace($arguments, $displayArgs) }
+
+    if ( $process.ExitCode -ne 0 )
+    {
+        Write-Host $stdout
+
+        $script:exitcode = $process.ExitCode
+        Write-Host -foregroundcolor red "ERROR: $exe exited with exit code " $process.ExitCode
+        if ($stderr) 
+        {
+            Write-Error "ERROR: $stderr"
+        }
+        throw "Failed to publish symbols"
+    }
+    else 
+    {
+        $stdout | Write-Verbose
+    }
+
+    if ( $stderr )
+    {
+        Write-Host -foregroundcolor red "ERROR: $stderr"
+    }
+
+    "$exe exited with exit code $($process.ExitCode)" | Write-Host
+
+    # Capture and convert JSON from stdout to make it look like we actually use the API
+    $stdout | where { $_ -match "({.+})" } | foreach { $matches[0] } | ConvertFrom-Json
+}
+
+function Unzip-SymbolClient([string]$clientZip, [string]$destinationDirectory)
+{
+    "Unzipping $($clientZip)" | Write-Verbose
+
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($clientZip, $destinationDirectory)
+
+    "Unzipped" | Write-Verbose
+}
+
+function Update-SymbolClient([string]$symbolServiceUri, [string]$symbolClientLocation)
+{
+    "Checking for most recent symbol.app.buildtask.zip version" | Write-Verbose
+
+    # Check latest package version.
+    $availableVersion = Get-SymbolClientVersion $symbolServiceUri
+
+    $clientPath = Join-Path $env:Temp $availableVersion
+    $symbolClientZip = Join-Path $clientPath "symbol.app.buildtask.zip"
+
+    if ( ! $(Test-Path $symbolClientZip -PathType Leaf) )
+    {
+        "$($symbolClientZip) not found" | Write-Verbose
+
+        if ( $(Test-Path $clientPath -PathType Container) )
+        {
+            "Cleaning $($clientPath)" | Write-Verbose
+
+            Remove-Item "$($clientPath)" -recurse | Write-Verbose
+        }
+
+        "Creating $($clientPath)" | Write-Verbose
+
+        New-Item -ItemType directory -Path $clientPath | Write-Verbose
+
+        $symbolClientZip = Download-SymbolClient $symbolServiceUri $clientPath
+    }
+    else
+    {
+        "$($symbolClientZip) exists" | Write-Verbose
+    }
+
+    $modulePath = Join-Path $clientPath "lib\net45"
+
+    if ( ! $(Test-Path $modulePath -PathType Container) )
+    {
+        Unzip-SymbolClient $symbolClientZip $clientPath | Write-Host
```

Comment:
another pattern to consider is writing a `.complete` file so you avoid a bad cached state.